provisioning/baremetal_secrets.go

```diff
@@ -161,3 +161,33 @@ password = %s
 	_, err = client.Secrets(targetNamespace).Create(context.Background(), secret, metav1.CreateOptions{})
 	return err
 }
+
+func DeleteMariadbPasswordSecret(client coreclientv1.SecretsGetter, targetNamespace string) error {
```

Comment:
why not make one function DeleteSecretIfExists(client, targetNamespace, targetName)

Reply:
Done.